packages/lodestar/src/chain/lightClient/index.ts

```diff
@@ -1 +1,579 @@
+   */
+  async pruneNonCheckpointData(nonCheckpointBlockRoots: Uint8Array[]): Promise<void> {
+    // TODO: Batch delete with native leveldb batching not just Promise.all()
+    await Promise.all([
+      this.db.genesisWitness.batchDelete(nonCheckpointBlockRoots),
+      this.db.syncCommitteeWitness.batchDelete(nonCheckpointBlockRoots),
+      this.db.finalizedHeader.batchDelete(nonCheckpointBlockRoots),
+    ]);
+  }
+
+  private async persistPostBlockImportData(
+    block: altair.BeaconBlock,
+    postState: CachedBeaconState<allForks.BeaconState>,
+    parentBlock: {blockRoot: RootHex; slot: Slot}
+  ): Promise<void> {
+    const blockSlot = block.slot;
+
+    const header: phase0.BeaconBlockHeader = {
+      slot: blockSlot,
+      proposerIndex: block.proposerIndex,
+      parentRoot: block.parentRoot,
+      stateRoot: block.stateRoot,
+      bodyRoot: this.config.getForkTypes(blockSlot).BeaconBlockBody.hashTreeRoot(block.body),
+    };
+
+    const blockRoot = ssz.phase0.BeaconBlockHeader.hashTreeRoot(header);
+    const blockRootHex = toHexString(blockRoot);
+
+    // Store genesis witness in case this block becomes a checkpoint
+    const genesisWitness = getGenesisWitness(postState);
+    await this.db.genesisWitness.put(blockRoot, genesisWitness);
+
+    const syncCommitteeWitness = getSyncCommitteesWitness(postState);
+
+    // Only store current sync committee once per run
+    if (!this.storedCurrentSyncCommittee) {
+      await this.storeSyncCommittee(postState.currentSyncCommittee, syncCommitteeWitness.currentSyncCommitteeRoot);
+      await this.storeSyncCommittee(postState.nextSyncCommittee, syncCommitteeWitness.nextSyncCommitteeRoot);
+      this.storedCurrentSyncCommittee = true;
+      this.logger.debug("Stored currentSyncCommittee", {slot: blockSlot});
+    }
+
+    // Only store next sync committee once per dependant root
+    const parentBlockPeriod = computeSyncPeriodAtSlot(parentBlock.slot);
+    const period = computeSyncPeriodAtSlot(blockSlot);
+    if (parentBlockPeriod < period) {
+      // If the parentBlock is in a previous epoch it must be the dependantRoot of this epoch transition
+      const dependantRoot = parentBlock.blockRoot;
+      const periodDependantRoots = this.knownSyncCommittee.getOrDefault(period);
+      if (!periodDependantRoots.has(dependantRoot)) {
+        periodDependantRoots.add(dependantRoot);
+        await this.storeSyncCommittee(postState.nextSyncCommittee, syncCommitteeWitness.nextSyncCommitteeRoot);
+        this.logger.debug("Stored nextSyncCommittee", {period, slot: blockSlot, dependantRoot});
+      }
+    }
+
+    // Ensure referenced syncCommittee are persisted before persiting this one
+    await this.db.syncCommitteeWitness.put(blockRoot, syncCommitteeWitness);
+
+    // Store header in case it is referenced latter by a future finalized checkpoint
+    await this.db.finalizedHeader.put(blockRoot, header);
```

Comment:
@dapplion could you double check this? I suppose we want to store the header of the finalized block instead?

Reply:
The naming here is the problem, this header is not finalized now but it may in the future. I'll rename to checkpointHeader